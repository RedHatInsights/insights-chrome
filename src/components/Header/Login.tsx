import React, { useContext } from 'react';
import { Button } from '@patternfly/react-core/dist/dynamic/components/Button';
import { useIntl } from 'react-intl';
import messages from '../../locales/Messages';
import ChromeAuthContext from '../../auth/ChromeAuthContext';

const Login = () => {
  const intl = useIntl();
  const { login } = useContext(ChromeAuthContext);
  return (
    <Button
      ouiaId="top-right-login-button"
      variant="tertiary"
      aria-label="Toggle primary navigation"
      widget-type="InsightsNavToggle"
      onClick={() => login()}
    >
      {intl.formatMessage(messages.login)}
    </Button>
  );
};

export default Login;
