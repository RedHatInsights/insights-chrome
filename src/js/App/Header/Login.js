import React from 'react';
import { Button } from '@patternfly/react-core';
import { login } from '../../jwt/jwt';

const Login = () => (
    <div className="pf-c-page__header-tools">
        <Button
            variant="tertiary"
            aria-label="Toggle primary navigation"
            widget-type="InsightsNavToggle"
            onClick={ login }>
          Log in
        </Button>
    </div>
);

export default Login;
