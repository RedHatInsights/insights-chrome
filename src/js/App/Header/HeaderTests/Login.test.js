import React from 'react';
import { render } from '@testing-library/react';
import Login from '../Login';

describe('Login', () => {
  it('should render correctly', () => {
    const { container } = render(<Login />);
    expect(container.querySelector('div')).toMatchSnapshot();
  });
});
