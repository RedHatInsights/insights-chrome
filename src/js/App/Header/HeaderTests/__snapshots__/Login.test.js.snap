// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div
  class="pf-c-page__header-tools"
>
  <button
    aria-disabled="false"
    aria-label="Toggle primary navigation"
    class="pf-c-button pf-m-tertiary"
    data-ouia-component-id="top-right-login-button"
    data-ouia-component-type="PF4/Button"
    data-ouia-safe="true"
    type="button"
    widget-type="InsightsNavToggle"
  >
    Log in
  </button>
</div>
`;
